lib/data_pull.rb

```diff
@@ -0,0 +1,224 @@
+require 'optparse'
+
+class DataPull
+  attr_reader :argv, :stdout, :stderr
+
+  def initialize(argv:, stdout:, stderr:)
+    @argv = argv
+    @stdout = stdout
+    @stderr = stderr
+  end
+
+  Result = Struct.new(
+    :table, # tabular output, rendered as an ASCII table or as CSV
+    :log_message, # summary message used for audit logging, DO NOT PUT PII HERE
+    keyword_init: true,
+  )
+
+  Config = Struct.new(
+    :include_missing,
+    :format,
+    :show_help,
+    keyword_init: true,
+  ) do
+    alias_method :include_missing?, :include_missing
+    alias_method :show_help?, :show_help
+  end
+
+  def config
+    @config ||= Config.new(
+      include_missing: true,
+      format: :table,
+      show_help: false,
+    )
+  end
+
+  def run
+    option_parser.parse!(argv)
+    subtask_class = subtask(argv.shift)
+
+    if config.show_help? || !subtask_class
+      stdout.puts option_parser
+      return
+    end
+
+    result = subtask_class.new.run(args: argv, include_missing: config.include_missing?)
+
+    stderr.puts result.log_message
+
+    render_output(result.table)
+  end
+
+  # @param [Array<Array<String>>] rows
+  def render_output(rows)
+    return if rows.blank?
+
+    case config.format
+    when :table
+      require 'terminal-table'
+      table = Terminal::Table.new
+      header, *body = rows
+      table << header
+      table << :separator
+      body.each do |row|
+        table << row
+      end
+      stdout.puts table
+    when :csv
+      require 'csv'
+      CSV.instance(stdout) do |csv|
+        rows.each do |row|
+          csv << row
+        end
+      end
+    when :json
+      headers, *body = rows
+
+      objects = body.map do |values|
+        headers.zip(values).to_h
+      end
+
+      stdout.puts JSON.pretty_generate(objects)
+    else
+      raise "Unknown format=#{config.format}"
+    end
+  end
+
+  # @api private
+  # A subtask is a class that has a run method, the type signature should look like:
+  # +#run(args: Array<String>, include_missing: Boolean) -> Result+
+  # @return [Class,nil]
+  def subtask(name)
+    {
+      'uuid-lookup' => UuidLookup,
+      'uuid-convert' => UuidConvert,
+      'email-lookup' => EmailLookup,
+      'profile-status' => ProfileStatus,
+    }[name]
+  end
```

Comment:
I feel like having these all in the same file is fine for now? Open to feedback if others feel differently